### PR TITLE
Make Python operations stricter about types

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
### Description

Fixes https://github.com/source-academy/py-slang/issues/39

### Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### How to test

The example given in the issue should throw error now.

### Checklist

- [ ] I have tested this code

### CODE REVIEW

**Comment by RichDom2185** [comment]
The tests seem to be failing?

**Comment by Fidget-Spinner** [comment]
I think the tests should pass now

File: src/stdlib/pylib.ts [hunk 1]
```diff
@@ -62,14 +74,20 @@ export function __py_multiplier(x: Value, y: Value) {
   if (typeof x === 'number' && typeof y === 'bigint') {
     return x * Number(y)
   }
-  return NaN
+  if (typeof x == 'number' && __is_string(y)) {
+    return y.repeat(x)
+  }
+  if (typeof y == 'number' && __is_string(x)) {
+    return x.repeat(y)
+  }
```
Comment (RichDom2185) [inline src/stdlib/pylib.ts:82]: These will never run because errors will be thrown above when `__is_numeric` gives false right?

**Review by RichDom2185 — COMMENTED** [review]
Hi, just the one comment, thanks!

File: src/stdlib/pylib.ts (hunk 1 above)
Comment (Fidget-Spinner) [reply src/stdlib/pylib.ts:82]: Oops you're right. Need to fix that.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Comment by Fidget-Spinner** [comment]
I need an approving review to merge it seems.

**Comment by RichDom2185** [comment]
> I need an approving review to merge it seems.

The CI turned out to fall, that's why I dismissed my approval

**Comment by Fidget-Spinner** [comment]
@RichDom2185 CI passes now.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**